components/raftstore/src/store/peer.rs

```diff
@@ -3118,8 +3144,6 @@ where
         if self.is_applying_snapshot()
             || self.has_pending_snapshot()
             || msg.get_from() != self.leader_id()
-            // For followers whose disk is full.
-            || disk::disk_full_precheck(ctx.store_id()) || ctx.is_disk_full
```

Comment:
I think this strategy is better:
1. don't transfer leadership to a disk almost full target, if the current leader is far from disk full;
2. transfer leadership to a disk almost full target, if the current leader is also almost full or already full
3. never transfer leadership to a target which is already disk full.